numpy/ma/testutils.py

```diff
@@ -126,7 +126,8 @@ def assert_equal(actual, desired, err_msg=''):
     if not (isinstance(actual, ndarray) or isinstance(desired, ndarray)):
         msg = build_err_msg([actual, desired], err_msg,)
         if not desired == actual:
-            raise AssertionError(msg)
+            # Handle case using testing.assert_equal
+            return np.testing.assert_equal(actual, desired)
```

Comment:
This doesn't seem right. Why build the error message and do the check, only to defer to the base-class check? Also: What happens if the order of the arguments is reversed?